src/gui/src/viewer/zoom-window.cpp

```diff
@@ -74,6 +75,44 @@ ZoomWindow::ZoomWindow(QList<QSharedPointer<Image>> images, const QSharedPointer
 	QShortcut *toggleFullscreen = new QShortcut(QKeySequence::FullScreen, this);
 		connect(toggleFullscreen, &QShortcut::activated, this, &ZoomWindow::toggleFullScreen);
 
+	
+	switch (m_settings->value("actionButtons").toInt()) {	// Consider adding this as class variable.
+		case ButtonVisibility::All :	// Handled later by resizeButtons.
+			break;
+		case ButtonVisibility::None :
+			ui->buttonPlus->setVisible(false);
+			ui->actionButtons->setVisible(false);
+			break;
+		case ButtonVisibility::Favorites :
+			ui->drawer_1->setVisible(true);
```

Comment:
"drawer_1" is not a very clear name, but I guess that's the parent for all favorites-related buttons?